python/ray/_raylet.pyx

```diff
@@ -1722,6 +1727,21 @@ cdef class CoreWorker:
             self.job_config.ParseFromString(c_job_config.SerializeAsString())
         return self.job_config
 
+    def prepare_runtime_env(self, serialized_runtime_env: str) -> str:
+        """Update parent's runtime env with new env via a simple dict update.
+
+        If the resulting runtime env is empty, fall back to the runtime env
+        set in the JobConfig.
+        """
+        runtime_env_dict = json.loads(self.current_serialized_runtime_env()
+                                      or "{}")
+        incoming_runtime_env_dict = json.loads(serialized_runtime_env or "{}")
+        runtime_env_dict.update(incoming_runtime_env_dict)
+        new_serialized_env = json.dumps(runtime_env_dict)
+        if new_serialized_env == "{}":
+            new_serialized_env = self.get_job_config().serialized_runtime_env
```

Comment:
It's a little weird that we have to do this at the end instead of short-circuiting. Is it possible to add a check that self.runtime_env and incoming_runtime_env are both None and then just return the job config's env? That will be the common case and we can avoid all of this logic & ser/de